File: LC#242-Valid_Anagram/valid_anagram.rs
/// 242. Valid Anagram
/// Difficulty: Easy
/// Topics: Hash Table, String, Sorting
/// https://leetcode.com/problems/valid-anagram/

fn main() {
    fn valid_anagram(s: String, t: String) -> bool {
        let mut char_count: [i32; 26] = [0; 26];
        if s.len() != t.len() {
            return false
        }

        for (&s_byte, &t_byte) in s.as_bytes().iter().zip(t.as_bytes().iter()) {
            char_count[(s_byte - b'a') as usize] += 1;
            char_count[(t_byte - b'a') as usize] -= 1;
        }

        char_count.iter().all(|&freq| freq == 0)
    }
    
    // Test cases
    let case_1_s = "anagram".to_string();
    let case_1_t = "nagaram".to_string();

    let case_2_s = "rat".to_string();
    let case_2_t = "car".to_string();
    
    assert_eq!(valid_anagram(case_1_s, case_1_t), true);
    assert_eq!(valid_anagram(case_2_s, case_2_t), false);
    
    println!("All tests passed successfully!");
}

File: Arrays_and_Hashing/LC#242-Valid_Anagram/valid_anagram.py
# 242. Valid Anagram
# Difficulty: Easy
# Topics: Hash Table, String, Sorting
# https://leetcode.com/problems/valid-anagram/

class Solution:
    def validAnagram(self, s: str, t: str) -> bool:
        char_count = [0] * 26

        if len(s) != len(t):
            return False
        
        for i in range(len(s)):
            char_count[ord(s[i]) - ord('a')] += 1
            char_count[ord(t[i]) - ord('a')] -= 1
        
        for char_freq in char_count:
            if char_freq != 0:
                return False
        return True
            
solution_instance = Solution()

# Test cases
case_1_s = "anagram"
case_1_t = "nagaram"

case_2_s = "rat"
case_2_t = "car"

assert solution_instance.validAnagram(case_1_s, case_1_t)
assert not solution_instance.validAnagram(case_2_s, case_2_t)

print("All tests passed successfully!")